### DISCUSSION TITLE
Things that we miss in BashOnWindows

### DISCUSSION THREAD

**Discussion by Zeioth** [microsoft/WSL#6916] [Q&A]
Hi, we are a team who work using node.js. We are very excited about _BashOnWindows_ because using it, we can write bash scripts to setup, and start our development environment in any OS. There is only two things that we miss in order to use it efectively:

  * **Launch new terminals:** Since there is no graphic server, We can't run commands in a new terminal. For example, I can't write a script that starts 'webpack-dev-server' and 'meteor'. In practice, this is a huge limitation.

* **Lack of file exposure:** If you copy a file from Windows to BashOnWindows using the file explorer, you won't be able to reach it on bash. The only way to achieve that, is with a bash command like:

    cp /mnt/c/path/of/a/windows/file /home/user/file

In practice, that means that my team don't have a graphical file explorer. They find it a very frustrating way to work. That's why we've decided stop using BashOnWindows by now. Hopefully our experience will help you to improve (or give us some hints). Feel free to ask us anything. 

Thank you for all your efforts.
Adrian.

**Comment by fpqc** [comment]
@Zeioth both of these are duplicates and both are being worked on.

**Comment by therealkenc** [comment]
> Since there is no graphic server, We can't run commands in a new terminal. For example, I can't write a script that starts 'webpack-dev-server' and 'meteor'.

Wait, webpak servers don't (or shouldn't) have anything to do with lack of a graphic server. Web clients and web servers speak http, which WSL does fine. You can run firefox or chrome with VcXsrv, which is not an endorsed scenario, or you can run the Windows web clients via interop, which is. But either way a https _server_ on WSL is entirely supported, and does not even require interop. The web client can be in another country let alone running on a native X11 server on WSL. Or I am misunderstanding your ask.

>  If you copy a file from Windows to BashOnWindows using the file explorer, you won't be able to reach it on bash. 

Working on it.

**Comment by mobluse** [comment]
You could use mc (Midnight Commander).
